flink-core/src/test/java/org/apache/flink/api/dag/TransformationTest.java

```diff
@@ -125,19 +145,34 @@ public void testDeclareManagedMemorySlotScopeUseCaseFailWrongScope() {
 
     /** A test implementation of {@link Transformation}. */
     private static class TestTransformation<T> extends Transformation<T> {
-
-        public TestTransformation(String name, TypeInformation<T> outputType, int parallelism) {
```

Comment:
It's better to introduce a `TestMultipleInputTransformation` which inherits `AbstractMultipleInputTransformation`.
Otherwise we are testing some testing logic instead of production logic.